processor/src/main/java/com/linecorp/decaton/processor/runtime/AveragingRateLimiter.java

```diff
@@ -55,14 +55,13 @@
 
     @Override
     public long acquire(int permits) throws InterruptedException {
-        if (latch.getCount() == 0) {
-            throw new IllegalStateException("This limiter is already closed");
-        }
-
         long microsToWait = reserve(permits);
         if (microsToWait > 0L) {
             latch.await(microsToWait, MICROSECONDS);
         }
+        if (terminated()) {
```

Comment:
It's a spec declared in interface 
<img width="894" alt="image" src="https://user-images.githubusercontent.com/2255161/78537213-1258f200-782a-11ea-8c7a-5a4771c83a1f.png">

The reason I thought that spec is better than returning partially awaited time is because those return values typically used to record metrics/logs of how long did it actually blocked, which is pointless or cases after `#close()` because callers are supposed to return immediately without proceeding to the next step after the limiter returns.

Reply:
understood.